Remove llvm toolchain supplied compiler options

kent-mcleod opened the pull request:
This will require: https://github.com/seL4/sel4runtime/pull/15 and https://github.com/seL4/musllibc/pull/14

Also will need to see if all the configurations compile and run because there tends to be a ton of edge cases when altering these flags.

- `-fno-integrated-as` prevents the llvm assembler from being.  The reason for this historically was because using llvm-as caused assembly errors.  Hopefully https://github.com/seL4/musllibc/pull/14/commits/8b644601dc98592817c5b3e09c6c0d26539ff961 is the only one left now.
- `-Wno-unused-command-line-argument` and `-Qunused-arguments` suppressed warnings about unused command line arguments. https://github.com/seL4/musllibc/pull/14/commits/88403f7404bbf79dff470cc771b6dba31d63b4b1 suppresses these warnings for muslc and https://github.com/seL4/sel4runtime/pull/15 and https://github.com/seL4/seL4/commit/f1d2e919567e7ecc155e16e44b3ee4c53bc45704 should resolve the rest.
- `-Wno-sizeof-pointer-div` and `-Wno-constant-logical-operand` should be applied more directly to the warnings that they are trying to suppress.  I haven't found any of these warnings yet.
- `-fno-experimental-isel` I haven't been able to reproduce this using clang-11. 

Test with seL4/sel4runtime#15 and seL4/musllibc#15

lsf37 commented:
There still seems to be something not quite in sync:

```
[67/295] Invoking muslc build system
FAILED: apps/sel4test-driver/musllibc/build-temp/stage/lib/libc.a
cd /github/workspace/build/apps/sel4test-driver/musllibc && rm -r build-temp && mkdir -p build-temp && /usr/bin/cmake -E env TOOLPREFIX=aarch64-linux-gnu- C_COMPILER=/usr/bin/clang MAKEFLAGS="${MAKEFLAGS:--s}" NK_CFLAGS="-nostdinc -fno-pic -fno-pie -fno-stack-protector -fno-asynchronous-unwind-tables -ftls-model=local-exec -mstrict-align -march=armv8-a -D__KERNEL_64__ --target=aarch64-linux-gnu" TARGET=aarch64 make -C build-temp -j4 -f /github/workspace/projects/musllibc/Makefile STAGE_DIR=/github/workspace/build/apps/sel4test-driver/musllibc/build-temp/stage SOURCE_DIR=/github/workspace/projects/musllibc && mkdir -p /github/workspace/.sel4_cache/musllibc/141a20ea315f4c65305bd1c85749c229 && tar -zcf code.tar.gz -C /github/workspace/build/apps/sel4test-driver/musllibc/build-temp/stage . && mv code.tar.gz /github/workspace/.sel4_cache/musllibc/141a20ea315f4c65305bd1c85749c229/
/github/workspace/projects/musllibc/src/internal/aarch64_sel4/syscall.s:10:5: error: unknown directive
1:  .dword __sysinfo-1b
    ^
```

kent-mcleod commented:
> There still seems to be something not quite in sync:

Ok, it's because the clang-8 version used in docker doesn't support that directive yet.  It was added at some point: https://github.com/llvm-mirror/llvm/commit/2deb068b06a478f446fdb980628cb2198f1fccfa#diff-6559010013f4d929da731794d1e2ace28a742d7386c12edb972d7dcdd4bfcbb5R250

I've changed that part of musl, which is code we added originally, to use a more generic directive: https://github.com/seL4/musllibc/pull/15

lsf37 reviewed (approved):
With that combination tests have now passed.
